api/routes/generate.js

```diff
@@ -47,149 +42,189 @@ exports.buildCrxConfig = function *(targetUrl) {
         filename: parsedUrl.hostname + '.crx'
     };
 
-    // Prepare request (send fake browser user-agent header)
-    var req = {
-        url: crxConfig.url,
-        headers: {
-            'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36'
-        }
-    };
-    
-    // Page HTML response
-    var res;
-    
+    var dom;
+
     // Execute GET request to provided URL
     // May fail for internal URLs, continue anyway
     try {
-        res = yield thunkify(request)(req);
+        dom = yield exports.getPageDOM(crxConfig.url);
     }
-    catch(exc) {
-        // Return fallback config
-        return crxConfig;
+    catch (exc) {
+        console.log("Failed to get page dom")
```

Comment:
As far as I know the best practice is to not have more than one return statement per function.
Changed it back because I know you will not allow that...